pkg/icons/icons.go

```diff
@@ -0,0 +1,144 @@
+package icons
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const (
+	partnerFilePath     = "index.yaml"
+	partnerDownloadPath = "assets/logos"
+)
+
+// PackageIconOverride is a struct to hold the package name, path, and icon to override
+type PackageIconOverride struct {
+	Name string
+	Path string
+	Icon string
+}
+
+// PackageIconList is a list of PackageIconOverride
+type PackageIconList []PackageIconOverride
+
+// PackageIconMap is a map of PackageIconOverride
+type PackageIconMap map[string]PackageIconOverride
+
+// ParsePackageToOverride will convert a Package from a PackageList to a PackageIcon for a PackageIconList
+func ParsePackageToOverride(name, path, icon string) PackageIconOverride {
+	return PackageIconOverride{
+		Name: name,
+		Path: path,
+		Icon: icon,
+	}
+}
+
+// CheckFilesStructure will check if index.yaml and assets/logos exist
+// will log and exit if the file structure does not exist
+func CheckFilesStructure() {
+	if _, err := os.Stat(partnerFilePath); os.IsNotExist(err) {
```

Comment:
This is just checking if we have the partner charts repository in the proper relative path, is a different check from checking if the icon file was downloaded.

Reply:
Oh, I was talking about how we are checking the existence of files. We could use the `Exists()` function here instead of `os.Stat()`. But if you want to leave this as it is, that is fine too.

Reply:
Fixed